launch_ros/launch_ros/actions/node.py

```diff
@@ -365,7 +367,24 @@ def execute(self, context: LaunchContext) -> Optional[List[Action]]:
                     '{}:={}'.format(remapping_from, remapping_to)
                 )
         context.extend_locals({'ros_specific_arguments': ros_specific_arguments})
-        return super().execute(context)
+        ret = super().execute(context)
+
+        execute_process_logger = launch.logging.get_logger(self.name)
+        if '<node_name_unspecified>' in self.node_name:
+            add_node_name(context, None)
+            execute_process_logger.warning(
+                'A node without an explicit name is being launched, '
+                'this may have the unintended side effect of non-unique node names. '
+                'It is recommended to explicitly name all nodes in launch files.')
+        else:
```

Comment:
I would just drop all the `if` case, and check for `if not ...`.

Reply:
Done.